pandas/core/generic.py

```diff
@@ -2841,14 +2850,25 @@ def clip_upper(self, threshold):
         -------
         clipped : same type as input
         """
-        if isnull(threshold):
+        if np.any(isnull(threshold)):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
-        return self.where((self <= threshold) | isnull(self), threshold)
+        if axis is None:
```

Comment:
hmm, I think you can simply use the else clause, right? (and just pass in axis as is)